eos/singularity_eos.f90

```diff
@@ -78,33 +80,33 @@ end function init_sg_Gruneisen
   
   interface
     integer(kind=c_int) function &
-      init_sg_JWL(matid, eos, A, B, R1, R2, w, rho0, Cv) &
+      init_sg_JWL(matindex, eos, A, B, R1, R2, w, rho0, Cv) &
       bind(C, name='init_sg_JWL')
       import
-      integer(c_int), value, intent(in)      :: matid
+      integer(c_int), value, intent(in)      :: matindex
       type(c_ptr), value, intent(in)         :: eos
       real(kind=c_double), value, intent(in) :: A, B, R1, R2, w, rho0, Cv
     end function init_sg_JWL
   end interface
 
   interface
     integer(kind=c_int) function &
-      init_sg_DavisProducts(matid, eos, a, b, k, n, vc, pc, Cv, E0) &
+      init_sg_DavisProducts(matindex, eos, a, b, k, n, vc, pc, Cv, E0) &
       bind(C, name='init_sg_DavisProducts')
       import
-      integer(c_int), value, intent(in)      :: matid
+      integer(c_int), value, intent(in)      :: matindex
       type(c_ptr), value, intent(in)         :: eos
       real(kind=c_double), value, intent(in) :: a, b, k, n, vc, pc, Cv, E0
     end function init_sg_DavisProducts
   end interface
 
   interface
     integer(kind=c_int) function &
-      init_sg_DavisReactants(matid, eos, rho0, e0, P0, T0, A, B, C, G0, Z,&
+      init_sg_DavisReactants(matindex, eos, rho0, e0, P0, T0, A, B, C, G0, Z,&
                           alpha, Cv0) &
       bind(C, name='init_sg_DavisReactants')
       import
-      integer(c_int), value, intent(in)      :: matid
+      integer(c_int), value, intent(in)      :: matindex
```

Comment:
Why this rename?

Reply:
I make sure to test with/without EOSPAC enabled. One complicating factor could be that I will want eospac to install the sesame2spiner executable but then not want the model to be available for the GPU build of the library. I guess as long as there are appropriate caveats and/or protections this should be ok, assuming I that combo is buildable and the compiler doesn't error out.